.github/workflows/workflow.yaml

```diff
@@ -39,25 +39,38 @@ jobs:
 #          mypy .
 
   test:
-    name: Test package
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    env:
+      REPORT: true
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python-version: [3.6, 3.7, 3.8]
+        include:
+          - os: ubuntu-latest
+            path: ~/.cache/pip
+          - os: macos-latest
+            path: ~/Library/Caches/pip
+        exclude:
+          - os: macos-latest
+            python-version: '3.8'
     steps:
-      - name: Checkout
-        uses: actions/checkout@v2
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
         with:
-          lfs: true
-          fetch-depth: 1
-
-      - name: Set up Python 3.7
-        uses: actions/setup-python@v1
+          python-version: ${{ matrix.python-version }}
+      - uses: actions/cache@v2
         with:
-          python-version: "3.7"
-
-      - name: Run unit tests
+          path: ${{ matrix.path }}
+          key: ${{ runner.os }}-pip-${{ hashFiles('reqs/base.txt') }}
```

Comment:
@carlomazzaferro One thing to note in the caching process for the testing matrix is the hash on 'reqs/base.txt'. I think it'll only establish a new cache when reqs/base.txt changes. So if any of the dependencies in reqs/extras change it may not install the new dependencies. What are your thoughts on this? Also, want to check that I am thinking about this correctly :)

Reply:
The overall approach looks sound to me (I'm still fairly new to GHA caching) -- perhaps `${{ hashFiles('setup.py','reqs/all.txt') }}` will provide the behaviour we want. It will update the cache when any dep is changed, or when `setup.py` changes (when bumping a new scikit-hts version). The latter is not strictly necessary, but probably a good practice. Wdyt?

Reply:
I like it, it makes sense to me. Making change